assembl/static2/js/app/components/common/ideaPreview.jsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { Link } from 'react-router';
+import { Translate } from 'react-redux-i18n';
+import Statistic from './statistic';
+
+class IdeaPreview extends React.Component {
```

Comment:
Functional component you mean? This is different. Only React.PureComponent or React.Component implementing shouldComponentUpdate, or using connect() are pure components.